plenum/server/replica.py

```diff
@@ -552,7 +552,12 @@ def _setup_for_non_master(self):
                 else lowest_prepared - 1
             self.logger.debug('{} Setting last ordered for non-master as {}'.format(
                 self, self.last_ordered_3pc))
-            self.last_ordered_3pc = (self.viewNo, lowest_ordered)
+            # now after catch up we have in last_ordered_3pc value from
+            # master replica and should change last_ordered_3pc
+            # to lowest_ordered or 0
+            self.last_ordered_3pc = (self.viewNo, lowest_ordered) \
+                if is_catchup or lowest_ordered > self.last_ordered_3pc[1] \
+                else self.last_ordered_3pc
             self._clear_last_view_message_for_non_master(self.viewNo)
```

Comment:
I think it makes sense to move this line out of the method and call it only on Primay Selection (when View Change ends).

Reply:
Why? If we changed last_ordered_3pc in _setup_for_non_master() then we need to clean stash for backup repicas because it's not important data (it can be dangerous for case with overtaking nodes) for us.

Reply:
But the method doesn't check anything related to `last_ordered_3PC`. The only parameter is `viewNo`.